api/handlers/listhandler.py

```diff
@@ -576,3 +584,132 @@ def packfile_end(self, cont_name, **kwargs):
         metadata = json.loads(self.request.GET.get('metadata'))
 
         return upload.process_upload(self.request, upload.Strategy.packfile, origin=self.origin, context={'token': token_id}, response=self.response, metadata=metadata)
+
+class AnalysesHandler(ListHandler):
+
+    def _check_ticket(self, ticket_id, _id, filename):
+        ticket = config.db.downloads.find_one({'_id': ticket_id})
+        if not ticket:
+            self.abort(404, 'no such ticket')
+        if ticket['ip'] != self.request.client_addr:
+            self.abort(400, 'ticket not for this source IP')
```

Comment:
Question: Is IP-matching of a ticket a new policy decision? If not, I would prefer we add it.
If it does exist, then this is good & let's keep it.